File: app/src/main/java/com/example/readative/room/entity/Review.java
package com.example.readative.room.entity;


import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Review {
    public static final int RATING_MIN = 1;
    public static final int RATING_MAX = 5;

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "review_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "book_owner_id")
    public long bookOwnerId;

    @NonNull
    @ColumnInfo(name = "review_rating")
    public int rating;
    @ColumnInfo(name = "review_content")
    public String content;

    public Review() {
        this.id = id;
        this.bookOwnerId = bookOwnerId;
        this.content = content;
        if (RATING_MIN <= rating && rating <= RATING_MAX) {
            this.rating = rating;
        } else if (RATING_MIN > rating) {
            this.rating = RATING_MIN;
        } else {
            this.rating = RATING_MAX;
        }
    }
}

File: app/src/main/java/com/example/readative/feature_book/data/data_source/models/Review.kt
package com.example.readative.feature_book.data.data_source.models

import androidx.room.*
import com.example.readative.feature_book.data.data_source.models.Book

@Entity(
    tableName = "reviews",
    indices = [
        Index(value = ["book_id"], unique = true)
    ],
    foreignKeys = [
        ForeignKey(
            entity = Book::class,
            parentColumns = ["id"],
            childColumns = ["book_id"],
            onUpdate = ForeignKey.CASCADE,
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Review(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "book_id") override val bookId: Long,
    @ColumnInfo(name = "rating") val rating: Int,
    @ColumnInfo(name = "text") val text: String
) : ReadativeEntity, BookIdEntity